mymodule/sky_sim.py

```diff
@@ -53,8 +81,10 @@ def make_positions(ra,dec, nsrc=NSRC):
     ras = []
     decs = []
     for _ in range(nsrc):
-        ras.append(ra + random.uniform(-1,1))
-        decs.append(dec + random.uniform(-1,1))
+        ras.append(ra + random.uniform(-1, 1))
+        decs.append(dec + random.uniform(-1, 1))
+    # apply our filter
+    ras, decs = crop_to_circle(ras, decs)
```

Comment:
This cropping will result in fewer sources, so that less than nsrc sources are returned